src/cmd/flux-job.c

```diff
@@ -1158,6 +1185,124 @@ void attach_stdin_cb (flux_reactor_t *r, flux_watcher_t *w,
     }
 }
 
+void MPIR_Breakpoint ()
+{
+    /* FIXME: Add this so that the compiler won't do funky things
+     * to this dummy function, in which case the debugger
+     * might have hard time setting a breakpoint here
+     */
+    printf ("MPIR_Breakpoint() is called\n");
+    printf ("MPIR_Breakpoint() is about to return\n");
+}
+
+static void valid_or_exit_for_debug (struct attach_ctx *ctx)
+{
+    int priority = 0;
+    uint32_t userid = 0;
+    flux_jobid_t id = 0;
+    double t_submit = 0.0f;
+    flux_future_t *f = NULL;
+    flux_job_state_t state = FLUX_JOB_INACTIVE;
+
+    if (!(f = flux_rpc_pack (ctx->h,
+                             "job-manager.info",
+                             FLUX_NODEID_ANY,
+                             0,
+                             "{s:I}",
+                             "id",
+                             ctx->id)))
+        log_err_exit ("flux_rpc_pack");
+
+    if (flux_rpc_get_unpack (f, "{s:I s:i s:i s:f s:i}",
+                                "id",
+                                &id,
+                                "userid",
+                                &userid,
+                                "priority",
+                                &priority,
+                                "t_submit",
+                                &t_submit,
+                                "state",
+                                &state) < 0)
+        log_err_exit ("Invalid job id (%ju) for debugging", (uintmax_t)id);
+
+    flux_future_destroy (f);
+
+    if (state > FLUX_JOB_RUN)
```

Comment:
Minor: Since only `state` is used in this function, you could change the `unpack` above to only unpack the `state`, i.e. you are not required to unpack all contents of a message if you don't need them. This also makes the unpack less sensitive to changes in the message payload.

Reply:
Also, as @garlick pointed out in #2635, this comparison might be a bit dangerous since a job state `> FLUX_JOB_RUN` could be added in the future which actually precedes the RUN state.

A better test here would to see ensure state is `FLUX_JOB_PENDING | FLUX_JOB_RUN`

Reply:
Good point. I can change this. 

I assumed the enum numeric values monotonically increases as a function of job lifecycle. (except for cyclic state transitions). Don't we have enough number spacing in the existing enum fields though? So that future states can also be added in lifecycle order?

Reply:
A good question. The states seem to be treated as flags so I'm not sure the intent there.
```C
typedef enum {
    FLUX_JOB_NEW                    = 1,
    FLUX_JOB_DEPEND                 = 2,
    FLUX_JOB_SCHED                  = 4,
    FLUX_JOB_RUN                    = 8,
    FLUX_JOB_CLEANUP                = 16,
    FLUX_JOB_INACTIVE               = 32,   // captive end state
} flux_job_state_t;
```
Maybe @garlick can comment on the design of `flux_job_state_t`

However, I don't think there is any guarantee of order implied, and in the future the state diagram may not be completely linear anyway. E.g. a STOP or similar state may be both before and after RUN.

Reply:
Yeah, like I said I'm fine with making this change.  

Going forward, I think we will have to have cycles with grow and shrink... It seems it is a good practice to have some "loose ordering" for readability though.

Reply:
They are consecutive powers of two (to make them maskable), so no actually.  Also just FYI but [RFC 21](https://github.com/flux-framework/rfc/blob/master/spec_21.adoc) says:
> The job state machine SHALL NOT contain cycles, to avoid ambiguity in synchronization.

Reply:
ooops sorry - I didn't see your other comments when I posted mine.   I thought we had talked through the issue of cycles and grow/shrink back when RFC 21 was landed.  I'll go see if there's a discussion over there somewhere and open an issue if not.

Reply:
Ah, thanks @garlick! I had forgotten about that rule in the RFC. And sorry about the dumb question about state masks (if I had looked at the virtual job states again I would have understood)